cmd/cape/cmd/token.go

```diff
@@ -41,9 +42,176 @@ var tokenCmd = &cobra.Command{
 	},
 }
 
+type createTokenReq struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+type tokenRef struct {
+	Name        string    `json:"name"`
+	Description string    `json:"description"`
+	CreatedAt   time.Time `json:"created_at"`
+}
+
+type listTokensResponse struct {
+	Tokens []tokenRef `json:"tokens"`
+}
+
+type createTokenResponse struct {
+	Token string `json:"token"`
+}
+
+var listTokensCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List your account tokens",
+	Long: `List the tokens from your account.
+
+NOTE -- this will not list function tokens (ones made with cape token <function_id>).
+This will be fixed in the future.`,
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		verbose, err := cmd.Flags().GetBool("verbose")
+		if err != nil {
+			return err
+		}
+
+		url := C.EnclaveHost
+		req, err := http.NewRequest(http.MethodGet, url+"/v1/tokens", nil)
+		if err != nil {
+			return err
+		}
+
+		authToken, err := getAuthToken()
+		if err != nil {
+			return err
+		}
+
+		var bearer = "Bearer " + authToken
+		req.Header.Add("Authorization", bearer)
+
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return err
+		}
+
+		if resp.StatusCode != http.StatusOK {
+			cmd.SilenceUsage = true
+			if verbose {
+				fmt.Println("received status code", resp.StatusCode)
+			}
+
+			return fmt.Errorf("something went wrong")
+		}
+
+		var ltr listTokensResponse
+		if err := json.NewDecoder(resp.Body).Decode(&ltr); err != nil {
+			return err
+		}
+
+		t := table.NewWriter()
+		t.SetOutputMirror(cmd.OutOrStdout())
+		t.AppendHeader(table.Row{"#", "Name", "Description", "Created At"})
+		localTime, err := time.LoadLocation("Local")
+		if err != nil {
+			return err
+		}
+
+		for i, token := range ltr.Tokens {
+			t.AppendRow([]interface{}{i, token.Name, token.Description, token.CreatedAt.In(localTime).Format("Jan 02 2006 15:04")})
+		}
+		t.SetStyle(table.StyleLight)
+		t.Render()
+
+		return nil
+	},
+}
+
+var createCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Create a token for your account",
+	Long: `Create a token for your account. 
+
+Use this command if you want a token that identifies you. 
+This is different than creating a token that gives access to an individual function.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		verbose, err := cmd.Flags().GetBool("verbose")
+		if err != nil {
+			return err
+		}
+
+		name, err := cmd.Flags().GetString("name")
+		if err != nil {
+			return err
+		}
+
+		description, err := cmd.Flags().GetString("description")
+		if err != nil {
+			return err
+		}
+
+		if name == "" {
+			return fmt.Errorf("tokens must be alphanumeric")
```

Comment:
Testing this out actually and found this message confusing

```suggestion
			return fmt.Errorf("token names must be alphanumeric")
```